ui/hooks/useMMICustodySignMessage.js

```diff
@@ -0,0 +1,72 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { showCustodianDeepLink } from '@metamask-institutional/extension';
+import { checkForUnapprovedTypedMessages } from '../store/institutional/institution-actions';
+import {
+  mmiActionsFactory,
+  setTypedMessageInProgress,
+} from '../store/institutional/institution-background';
+import {
+  accountsWithSendEtherInfoSelector,
+  getAccountType,
+  unapprovedTypedMessagesSelector,
+} from '../selectors';
+import { getAccountByAddress } from '../helpers/utils/util';
+import { getEnvironmentType } from '../../app/scripts/lib/util';
+import { goHome, showModal } from '../store/actions';
+import { ENVIRONMENT_TYPE_NOTIFICATION } from '../../shared/constants/app';
+
+export function useMMICustodySignMessage() {
+  const dispatch = useDispatch();
+  const mmiActions = mmiActionsFactory();
+  const envType = getEnvironmentType();
+  const accountType = useSelector(getAccountType);
+  const isNotification = envType === ENVIRONMENT_TYPE_NOTIFICATION;
+  const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
+  const unapprovedTypedMessages = useSelector(unapprovedTypedMessagesSelector);
+
+  const custodySignFn = async (_msgData) => {
+    if (accountType === 'custody') {
+      const { address: fromAddress } =
+        getAccountByAddress(allAccounts, _msgData.msgParams.from) || {};
+      try {
+        let msgData = _msgData;
+        let id = _msgData.custodyId;
+        if (!_msgData.custodyId) {
+          msgData = checkForUnapprovedTypedMessages(
+            _msgData,
+            unapprovedTypedMessages,
+          );
+          id = msgData.custodyId;
+        }
+        dispatch(
```

Comment:
Same as before. For some reason, `showCustodianDeepLink` on the previous `mapDispatchToProps` was not wrapped in a dispatch. It is worth asking the MMI team about it.